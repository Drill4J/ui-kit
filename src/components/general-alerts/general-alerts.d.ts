/// <reference types="react" />
declare type Message = 'SUCCESS' | 'ERROR' | 'WARNING' | 'INFO';
interface Props {
  className?: string;
  children?: React.ReactNode;
  type: Message;
}
export declare const GeneralAlerts: {
  ({ className, children, type }: Props): JSX.Element;
  displayName: string;
  __docgenInfo: {
    description: string;
    displayName: string;
    props: {
      className: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
      type: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
    };
  };
};
export {};
