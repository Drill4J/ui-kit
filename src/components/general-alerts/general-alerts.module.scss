@import '../../style-guide/index';

.general-alerts {
  display: flex;
  font-size: 14px;
  line-height: 20px;

  &__content {
    display: flex;
    width: 100%;
    padding: 12px 24px;
    word-break: break-word;

    &--type {
      &--success {
        background-color: map-get($green, light-tint);
      }

      &--error {
        background-color: map-get($red, ultralight-tint);
      }

      &--warning {
        background-color: map-get($orange, light-tint);
      }

      &--info {
        color: map-get($monochrome, default);
        background-color: map-get($monochrome, light-tint);
      }
    }
  }

  &__message-icon {
    display: flex;
    align-self: start;
    align-items: center;
    width: 20px;
    height: 20px;
    margin-right: 8px;

    &--type {
      &--success {
        color: map-get($green, default);
      }

      &--error {
        color: map-get($red, default);
      }

      &--warning {
        color: map-get($orange, default);
      }

      &--info {
        color: map-get($monochrome, default);
      }
    }
  }
}
