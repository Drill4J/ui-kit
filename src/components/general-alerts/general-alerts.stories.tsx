import { storiesOf } from '@storybook/react';

import { GeneralAlerts } from './general-alerts';
import { Spinner } from '../spinner';

storiesOf('GeneralAlerts', module)
  .add('SUCCESS', () => (
    <GeneralAlerts type="SUCCESS">
      Success message example. Something good happened!
    </GeneralAlerts>
  ))
  .add('ERROR with WARNING', () => (
    <>
      <GeneralAlerts type="ERROR" data-test="general-alerts:error">
        Error message example. Something bad happened.
        Error message example. Something bad happened.
      </GeneralAlerts>
      <GeneralAlerts type="WARNING" data-test="general-alerts:warning">
        Warning message example. Something affects product usage.
      </GeneralAlerts>
    </>
  ))
  .add('WARNING', () => (
    <GeneralAlerts type="WARNING">
      Warning message example. Something affects product usage.
    </GeneralAlerts>
  ))
  .add('INFO', () => (
    <GeneralAlerts type="INFO" />
  ))
  .add('WARNING alert with component', () => (
    <GeneralAlerts type="INFO">
      please wait...
      <Spinner />
    </GeneralAlerts>
  ));
