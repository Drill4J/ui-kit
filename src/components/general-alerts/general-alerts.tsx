import styled, { css } from 'styled-components';

import { Icons } from '../icon';
import { COLORS } from '../../theme';

type Message = 'SUCCESS' | 'ERROR' | 'WARNING' | 'INFO'

interface Props {
  className?: string;
  children?: React.ReactNode;
  type: Message;
}

export const GeneralAlerts = ({
  className, children = 'Internal service error', type,
}: Props) => (
  <Wrapper className={className}>
    <Content type={type}>
      {getMessageIcon(type)}
      {children}
    </Content>
  </Wrapper>
);

const Wrapper = styled.div`
  display: flex;
  font-size: 14px;
  line-height: 20px;
`;

const Content = styled.div<{type?: Message}>`
  display: flex;
  width: 100%;
  align-items: center;
  padding: 12px 24px;
  word-break: break-word;
  
  ${({ type }) => [
    type === 'SUCCESS' && css`background-color: ${COLORS.GREEN.LIGHT_TINT}`,
    type === 'ERROR' && css`background-color: ${COLORS.RED.ULTRALIGHT_TINT}`,
    type === 'INFO' && css`
      color: ${COLORS.MONOCHROME.DEFAULT};
      background-color: ${COLORS.MONOCHROME.LIGHT_TINT};
    `,
    type === 'WARNING' && css`background-color: ${COLORS.ORANGE.LIGHT_TINT}`,
  ]}
`;

const MessageIcon = styled.div<{type?: Message}>`
  display: flex;
  align-self: start;
  align-items: center;
  height: 20px;
  margin-right: 8px;
  
  ${({ type }) => [
    type === 'SUCCESS' && css`color: ${COLORS.GREEN.DEFAULT}`,
    type === 'ERROR' && css`color: ${COLORS.RED.DEFAULT}`,
    type === 'INFO' && css`color: ${COLORS.MONOCHROME.DEFAULT};`,
    type === 'WARNING' && css`color: ${COLORS.ORANGE.DEFAULT}`,
  ]}
`;

function getMessageIcon(type: Message) {
  switch (type) {
    case 'ERROR': {
      return (
        <MessageIcon type={type}>
          <Icons.Cancel height={16} width={16} />
        </MessageIcon>
      );
    }
    case 'SUCCESS': {
      return (
        <MessageIcon type={type}>
          <Icons.Checkbox height={16} width={16} />
        </MessageIcon>
      );
    }
    case 'WARNING': {
      return (
        <MessageIcon type={type}>
          <Icons.Warning height={16} width={16} />
        </MessageIcon>
      );
    }
    default: {
      return (
        <MessageIcon type={type}>
          <Icons.Info height={16} width={16} />
        </MessageIcon>
      );
    }
  }
}
