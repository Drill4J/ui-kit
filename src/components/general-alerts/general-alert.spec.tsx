import React from 'react';
import renderer from 'react-test-renderer';
import { GeneralAlerts } from './general-alerts';

const ALERT_MESSAGE = 'ALERT_MESSAGE';

describe('GeneralAlerts', () => {
  it('should match snapshot with type is ERROR', () => {
    const tree = renderer.create(<GeneralAlerts type="ERROR">{ALERT_MESSAGE}</GeneralAlerts>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with type is SUCCESS', () => {
    const tree = renderer.create(<GeneralAlerts type="SUCCESS">{ALERT_MESSAGE}</GeneralAlerts>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with type is WARNING', () => {
    const tree = renderer.create(<GeneralAlerts type="WARNING">{ALERT_MESSAGE}</GeneralAlerts>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with type is INFO', () => {
    const tree = renderer.create(<GeneralAlerts type="INFO">{ALERT_MESSAGE}</GeneralAlerts>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
