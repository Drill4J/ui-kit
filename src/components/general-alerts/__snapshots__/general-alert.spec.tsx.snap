// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralAlerts should match snapshot with type is ERROR 1`] = `
<div
  className="general-alerts___StyledDiv-sc-97788p-2 APiVU"
  data-test="general-alert"
>
  <div
    className="general-alerts__Content-sc-97788p-0 cQjFzV"
    data-test="general-alert:content"
    type="ERROR"
  >
    <div
      className="general-alerts__MessageIcon-sc-97788p-1 jYIdWF"
      type="ERROR"
    >
      <svg
        className="generic-icon__GenericIcon-sc-1g5t9jj-0 jPhxLI"
        fillRule="evenodd"
        height={16}
        style={
          Object {
            "WebkitTransform": "rotate(0deg)",
          }
        }
        transform="rotate(0)"
        viewBox="0 0 16 16"
        width={16}
      >
        <path
          d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zm0 .8a7.2 7.2 0 1 0 0 14.4A7.2 7.2 0 0 0 8 .8zM5.858 5.145L8 7.287l2.142-2.142a.5.5 0 0 1 .707 0l.006.006a.5.5 0 0 1 0 .707L8.713 8l2.142 2.142a.5.5 0 0 1 0 .707l-.006.006a.5.5 0 0 1-.707 0L8 8.713l-2.142 2.142a.5.5 0 0 1-.707 0l-.006-.006a.5.5 0 0 1 0-.707L7.287 8 5.145 5.858a.5.5 0 0 1 0-.707l.006-.006a.5.5 0 0 1 .707 0z"
        />
      </svg>
    </div>
    ALERT_MESSAGE
  </div>
</div>
`;

exports[`GeneralAlerts should match snapshot with type is INFO 1`] = `
<div
  className="general-alerts___StyledDiv-sc-97788p-2 APiVU"
  data-test="general-alert"
>
  <div
    className="general-alerts__Content-sc-97788p-0 GBfoY"
    data-test="general-alert:content"
    type="INFO"
  >
    <div
      className="general-alerts__MessageIcon-sc-97788p-1 kczcGV"
      type="INFO"
    >
      <svg
        className="generic-icon__GenericIcon-sc-1g5t9jj-0 jPhxLI"
        fillRule="evenodd"
        height={16}
        style={
          Object {
            "WebkitTransform": "rotate(0deg)",
          }
        }
        transform="rotate(0)"
        viewBox="0 0 20 20"
        width={16}
      >
        <path
          d="M 10 18.75 C 14.832 18.75 18.75 14.832 18.75 10 C 18.75 5.168 14.832 1.25 10 1.25 C 5.168 1.25 1.25 5.168 1.25 10 C 1.25 14.832 5.168 18.75 10 18.75 Z  M 10 20 C 4.477 20 0 15.523 0 10 C 0 4.477 4.477 0 10 0 C 15.523 0 20 4.477 20 10 C 20 15.523 15.523 20 10 20 Z  M 9.65 7.5 L 10.35 7.5 C 10.847 7.5 11.25 7.903 11.25 8.4 L 11.25 15.35 C 11.25 15.847 10.847 16.25 10.35 16.25 L 9.65 16.25 C 9.153 16.25 8.75 15.847 8.75 15.35 L 8.75 8.4 C 8.75 7.903 9.153 7.5 9.65 7.5 Z  M 9.65 3.75 L 10.35 3.75 C 10.847 3.75 11.25 4.153 11.25 4.65 L 11.25 5.35 C 11.25 5.847 10.847 6.25 10.35 6.25 L 9.65 6.25 C 9.153 6.25 8.75 5.847 8.75 5.35 L 8.75 4.65 C 8.75 4.153 9.153 3.75 9.65 3.75 Z"
        />
      </svg>
    </div>
    ALERT_MESSAGE
  </div>
</div>
`;

exports[`GeneralAlerts should match snapshot with type is SUCCESS 1`] = `
<div
  className="general-alerts___StyledDiv-sc-97788p-2 APiVU"
  data-test="general-alert"
>
  <div
    className="general-alerts__Content-sc-97788p-0 cLBIfY"
    data-test="general-alert:content"
    type="SUCCESS"
  >
    <div
      className="general-alerts__MessageIcon-sc-97788p-1 dhtoIh"
      type="SUCCESS"
    >
      <svg
        className="generic-icon__GenericIcon-sc-1g5t9jj-0 jPhxLI"
        fillRule="evenodd"
        height={16}
        style={
          Object {
            "WebkitTransform": "rotate(0deg)",
          }
        }
        transform="rotate(0)"
        viewBox="0 0 20 20"
        width={16}
      >
        <path
          d="M 10 1 C 5.029 1 1 5.029 1 10 C 1 14.971 5.029 19 10 19 C 14.971 19 19 14.971 19 10 C 19 5.029 14.971 1 10 1 Z  M 10 0 C 15.523 0 20 4.477 20 10 C 20 15.523 15.523 20 10 20 C 4.477 20 0 15.523 0 10 C 0 4.477 4.477 0 10 0 Z  M 4.697 9.944 L 4.803 9.834 C 5.189 9.433 5.812 9.434 6.196 9.833 L 7.804 11.501 C 8.19 11.901 8.808 11.903 9.188 11.508 L 13.812 6.714 C 14.194 6.317 14.812 6.324 15.197 6.723 L 15.303 6.832 C 15.689 7.233 15.69 7.877 15.308 8.273 L 9.192 14.616 C 8.813 15.009 8.19 15.011 7.807 14.614 L 4.693 11.386 C 4.309 10.987 4.312 10.343 4.697 9.944 Z "
        />
      </svg>
    </div>
    ALERT_MESSAGE
  </div>
</div>
`;

exports[`GeneralAlerts should match snapshot with type is WARNING 1`] = `
<div
  className="general-alerts___StyledDiv-sc-97788p-2 APiVU"
  data-test="general-alert"
>
  <div
    className="general-alerts__Content-sc-97788p-0 gOGjOg"
    data-test="general-alert:content"
    type="WARNING"
  >
    <div
      className="general-alerts__MessageIcon-sc-97788p-1 exRiVM"
      type="WARNING"
    >
      <svg
        className="generic-icon__GenericIcon-sc-1g5t9jj-0 jPhxLI"
        fillRule="evenodd"
        height={16}
        style={
          Object {
            "WebkitTransform": "rotate(0deg)",
          }
        }
        transform="rotate(0)"
        viewBox="0 0 16 16"
        width={16}
      >
        <path
          d="M8 16A8 8 0 1 1 8 0a8 8 0 0 1 0 16zm0-1A7 7 0 1 0 8 1a7 7 0 0 0 0 14zm0-6a1 1 0 0 1-1-1V5a1 1 0 0 1 2 0v3a1 1 0 0 1-1 1zm0 3a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
        />
      </svg>
    </div>
    ALERT_MESSAGE
  </div>
</div>
`;
